src/OpenSage.Game/Terrain/WaterArea.cs

```diff
@@ -44,16 +44,43 @@ public sealed class WaterArea : DisposableBase
             return true;
         }
 
-        private WaterArea(
+        internal static bool TryCreate(
             AssetLoadContext loadContext,
-            PolygonTrigger trigger)
+            StandingWaterArea area,
+            out WaterArea result)
         {
-            var triggerPoints = trigger.Points
-                .Select(x => new Vector2(x.X, x.Y))
-                .ToArray();
+            if (area.Points.Length < 3)
+            {
+                // Some maps (such as Training01) have water areas with fewer than 3 points.
+                result = null;
+                return false;
+            }
+
+            result = new WaterArea(loadContext, area);
+            return true;
+        }
 
+        internal static bool TryCreate(
+            AssetLoadContext loadContext,
+            StandingWaveArea area,
```

Comment:
We may end up wanting a different class for waves vs static water, but this is fine for now.